### DISCUSSION TITLE
Storytel reader android package

### DISCUSSION THREAD

**Discussion by raymon-m** [ntherning/StorytelReaderMods#13] [Q&A]
First, thanks for this cool project! It suddenly made an otherwise pretty useless device much more usable for me.

I will start by giving some context on what I did, in case it contains useful information for future reference. I have a Storyrel Reader V3 (2nd generation device).
I followed all your steps in the readme on my system running windows 11. I tried both the patched boot image you created for my gen2 device, as well as patching my own boot image. 
I started with the boot image you provided. Was able to flash it, install fdroid and run RelaunchX. But unfortunately got told by RelaunchX that I do not have root permission. Also did not see magisk installed in the launchable apps. At this point decided to try and patch my own image.
Straight off the bat, the python scripts gave me errors. It started with permission to copy some of the files, but it did seem to copy them nonetheless. Eventually I modified the script to not have it copy those files, and copied them manually. Then some errors were thrown where the cpiofile script attempted to access some os attributes that it couldn't. I had it skip those attributes so the script could run to completion, resulting in an image being created. Unfortunately flashing that one to the reader caused a boot loop. So I then replaced it with the image you created once more.
From then on I had a working device, having fdroid, relaunchx and koreader installed. But some of the books I purchased some time ago were only available in an app that you can only get through the play store. I decided to play around a bit with getting the play store installed, including trying some other google services to getting it to run. This resulted in endless crashes, never running stable for more than a few seconds. So I decided to abandon that idea for now and remove all the google packages I installed while trying this.

And this is where I messed up. I foolishly removed the google packages, including a storytel one, without making a backup first. Since then my device no longer responds to touch input. I suspect the package I removed but shouldn't have is com.storytel.inkreader, which I presume is responsible for handling the touch input. I am looking for someone who has a backup of their device that still has this package installed, that I could use to restore from. Or even simply has an apk for just this one package. I haven't tried reaching out to Storytel themselves yet, and any help is much appreciated!

**Comment by ntherning** [comment]
IIRC the inkreader APK is downloaded by the bootstrap app. Use the URL below to get a JSON with the URL of the latest inkreader APK under the `app.url` key:

https://api.storytel.net/reader-updates/devices/salamander/updates

The firmware file referenced by the JSON might also have useful files.

I do have backups of the raw partitions from both gen devices.  We can try that if you cannot get it to work with APK from the URL above.

**Reply by raymon-m** [reply]
Thanks a bunch! That did let me install the inkreader package. Unfortunately didnt fix touch input, so I will need to dive into that further.

**Comment by raymon-m** [comment]
So, coming back to this. I tried a bunch of things, including installing and re-installing a lot of APK's. None of that proved very helpful. Eventually I ended up opening up the device and re-seating all ribbon cables I could find to make sure there was no hardware failure instead. This also didn't seem to help as a reboot afterwards left me with no touch still. 
I then left it to rest for a few days, deciding to get back to it when I have a little more time.  Now suddenly, after reboot again, I have touch input. Based on this, I think it is more likely that this was a hardware issue and not a software one.
